neural_compressor/transformers/models/modeling_auto.py

```diff
@@ -503,27 +742,43 @@ def load_low_bit(cls, pretrained_model_name_or_path, *model_args, **kwargs):
             logger.warning("fp32 scale_dtype is used, please change the config.json if you don't want to use it.")
 
         # weight dtype is higher priority than bits in config.json when both existed.
-        if quantization_config.bits == 4:
-            if use_xpu:
-                quantization_config.weight_dtype = "int4_fullrange"
-            else:
-                quantization_config.weight_dtype = "int4"
-            logger.info(
-                "{} quantization weight_dtype is used due to bits is 4 in config.json.".format(
-                    quantization_config.weight_dtype
+        if quantization_config.weight_dtype is None:
+            if quantization_config.bits == 4:
+                if use_xpu:
+                    quantization_config.weight_dtype = "int4_fullrange"
+                else:
+                    quantization_config.weight_dtype = "int4_clip"
+                logger.info(
+                    "{} quantization weight_dtype is used due to bits is 4 in config.json.".format(
+                        quantization_config.weight_dtype
+                    )
                 )
-            )
-        elif quantization_config.bits == 8:
-            quantization_config.weight_dtype = "int8"
-            logger.info(
-                "{} quantization weight_dtype is used due to bits is 8 in config.json.".format(
-                    quantization_config.weight_dtype
+            elif quantization_config.bits == 8:
+                quantization_config.weight_dtype = "int8"
+                logger.info(
+                    "{} quantization weight_dtype is used due to bits is 8 in config.json.".format(
+                        quantization_config.weight_dtype
+                    )
+                )
+            else:
+                logger.warning("bits number only supports 4, 8.")
+                quantization_config.weight_dtype = "int4_clip"
+                logger.warning(
+                    "int4_clip weight_dtype is used, please change the config.json if you don't want to use it."
                 )
-            )
         else:
-            logger.warning("bits number only supports 4, 8.")
-            quantization_config.weight_dtype = "int4"
-            logger.warning("int4 weight_dtype is used, please change the config.json if you don't want to use it.")
+            if quantization_config.weight_dtype not in [
+                "int4_fullrange",
+                "int4_clip",
+                "int8",
+                "nf4",
+            ]:
+                logger.warning("Please provide the correct bits number or weight_dtype in config.json.")
+                raise ValueError(
+                    "weight_dtype must be a string in " "'int8', 'int4', 'int4_fullrange', 'int4_clip', 'nf4', "
```

Comment:
the same, please confirm IPEX kernel requested datatype is int4, int8 for cpu and int4_fullrange for xpu.